### PR TITLE
Change readiness probe to use metrics

### PR DESCRIPTION
**Pull request by thom-at-redhat**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/routes/imagesets.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// nolint:gocritic,govet,revive
```
Comment (ldjebran) [inline pkg/routes/imagesets.go:1]: Do not think this is a good idea putting them again , if they are failing on 1.19 and we are running 1.18 we have to find a solution for this, do not think this is how it should be, having this skips, on each file of the project.

Comment (thom-at-redhat) [reply pkg/routes/imagesets.go:1]: we are running against Go 1.18 starting with next patch

Comment (ldjebran) [reply pkg/routes/imagesets.go:1]: as per our offline discussion, this is an intermediate stage , resolve comment

**Review by ldjebran — APPROVED** [review]
lgtm

**Pull request merged**